### PR TITLE
Add defaults to propertyMap in PageSelectionResolver

### PR DESCRIPTION
**Pull request by alexandersch**

### CODE REVIEW

**Review by nnatter — COMMENTED** [review]
Would be great if you could adjust the test cases 🙂 

Also, I think we should add this to the `PageDataProviderResolver` too to keep thinks consistent

**Comment by alexandersch** [comment]
I will look into that in a bit.

File: Content/StructureResolver.php [hunk 1]
```diff
@@ -89,7 +89,7 @@ public function resolveProperties(
         array $propertyMap,
         string $locale,
         bool $includeExtension = false
-    ): array {
+    ): ?array {
```
Comment (alexander-schranz) [inline Content/StructureResolver.php:92]: this should not be nullable

File: Content/StructureResolverInterface.php [hunk 2]
```diff
@@ -36,5 +36,5 @@ public function resolveProperties(
         array $propertyMap,
         string $locale,
         bool $includeExtension = false
-    ): array;
+    ): ?array;
```
Comment (alexander-schranz) [inline Content/StructureResolverInterface.php:39]: this should not be nullable

**Review by alexander-schranz — CHANGES_REQUESTED** [review]
You should not make the structureResolver nullable.
What you are currently missing is that the expected calls in the test need to be changed as ` is now called always with title and url.

So the:

```
$this->structureResolver->resolveProperties()
```

calls need to be adjusted in the [PageSelectionResolverTest](https://github.com/sulu/SuluHeadlessBundle/blob/1bea2a10d52a1b062311fabb067dee71a422bb91/Tests/Unit/Content/ContentTypeResolver/PageSelectionResolverTest.php#L127)

because all now have `title` and `url` in it. For example:

```diff
        $this->structureResolver->resolveProperties(
            $pageStructure1->reveal(),
            [
                'title' => 'title',
+                'url' => 'url',
                'excerptTitle' => 'excerpt.title',
                'categories' => 'excerpt.categories',
            ],
            'en'
        )->willReturn([
            'id' => 'page-id-1',
            'template' => 'default',
            'content' => [
                'title' => 'Page Title 1',
                'excerptTitle' => 'Page Excerpt Title 1',
            ],
            'view' => [
                'title' => [],
                'excerptTitle' => [],
            ],
        ]);
```

if you adjust this the tests should be again green.

**Review by alexander-schranz — COMMENTED** [review]
@alexandersch I updated the tests for you. Now all should be fine. Thank you for your effort!

**Review by nnatter — APPROVED** [review]
Looks good - thanks a lot to both of you!

**Pull request merged**